packages/coinstac-common/src/models/pipeline/pipeline.js

```diff
@@ -212,40 +229,48 @@ class Pipeline extends Base {
   }
 
   /**
-   * Attempt to bump the pipeline step.  runs the `next` computation
-   * if present, and conditionally bumps the pipeline state. to be clear, this
-   * runs a computation's `next` computation, _not the next step in the pipeline_
+   * Maybe increment the pipeline's step.
    *
-   * if tryNext is called pre-`run` (`cycle.preRun`) and no `next` is defined,
-   * the pipeline will **not progress** its current computation index.
+   * This method increments the pipeline's `step` property based on a few
+   * conditions:
    *
-   * if tryNext is called post-`run` (`cycle.postRun`) and no `next` is defined,
-   * the pipeline **will progress** its current computation index.
+   * * If `isPreRun` is `true` and the computation has no `.next` property **do
+   *   not increment**.
+   * * If `isPreRun` is `false` and the computation has no `.next` property **do
+   *   increment**.
+   * * If a `.next` property exists pass `opts`. The `.next` computation should
+   *   return a boolean value indicating whether the pipeline should increment
+   *   its step.
+   * * Don't increment if there's no more computations to run.
    *
    * @private
-   * @param {object} opts inputs to next computation
-   * @param {object} cycle
-   * @param {boolean=} cycle.preRun tryNext is being called before `run` is called
-   * @param {boolean=} cycle.postRun tryNext is being called after `run` is called
-   * @returns {Promise}
+   * @param {Object} opts Inputs to next computation
+   * @param {boolean} [isPreRun=false] Whether method is called before
+   * `Pipeline#run` is called
+   * @returns {Promise<boolean>} Value representing whether the pipeline
+   * instance's `step` property was incremented and its `computation` property
+   * was updated: `true` indicates incrementation occurred, `false` indicates no
+   * change.
    */
-  tryNext(opts, cycle) {
-    const resp = { didStep: false };
+  maybeIncrementStep(opts, isPreRun) {
```

Comment:
`conditionallyIncrement`? dunno this function is kinda convoluted, makes it harder to put meaning in an easy name